### PR TITLE
Added layer merging

### LINKED ISSUE #120: Merge layers

### PR DESCRIPTION
**Pull request by CPKreuz**
Only thing that doesn't work is undoing, I could use some help with that @flabbet 

Closes #120

### CODE REVIEW

**Comment by flabbet** [comment]
Please write tests for your changes

File: PixiEditor/Models/DataHolders/Document.cs [hunk 1]
```diff
@@ -425,6 +426,74 @@ public void CenterContent()
...
+            foreach (Layer layer in layers)
+            {
+                Layers.Remove(layer);
+            }
+
+            SetActiveLayer(Layers.IndexOf(finalLayer));
+
+            return finalLayer;
+        }
+
```
Comment (flabbet) [inline PixiEditor/Models/DataHolders/Document.cs:496]: `Document.cs` is getting too long, maybe split some methods as other partial one or extension to `Document` class

**Review by flabbet — CHANGES_REQUESTED** [review]
I'll think of a method to implement undo, for now, add tests and change a thing I mentioned.

**Comment by CPKreuz** [comment]
> I'll think of a method to implement undo, for now, add tests and change a thing I mentioned.

Problem is that it crashes now when undoing

**Comment by flabbet** [comment]
I have an idea since we have combined layers, we need to store only BitmapPixelChanges of pixels that are gone due to merging, layer name, offset, width, height, and index, you need to save those pixels during merging

**Comment by flabbet** [comment]
Or what about saving the layers as a .pixi file in the temp folder and load it on demand?

**Review by flabbet — CHANGES_REQUESTED** [review]
I added undo to this, but I found one behavior thing that bothers me. If you merge with below, layers get merged, but in reversed order. I don't think it should work like that

**Comment by CPKreuz** [comment]
> I added undo to this, but I found one behavior thing that bothers me. If you merge with below, layers get merged, but in reversed order. I don't think it should work like that

ah shit, forgot I had to fix that

**Review by flabbet — APPROVED** [review]
_    
       | |   
  ___ | | __
 / _ \| |/ /
| (_) |   < 
 \___/|_|\_\

**Pull request merged**